identity/aziot-dps-client-async/src/lib.rs

```diff
@@ -70,68 +87,130 @@ impl Client {
             self.scope_id, registration_id
         );
 
-        let body = model::DeviceRegistration {
-            registration_id: Some(registration_id.into()),
+        let body = match auth_kind {
+            DpsAuthKind::Tpm => {
+                let aziot_tpm_common::TpmKeys {
+                    endorsement_key,
+                    storage_root_key,
+                } = self.tpm_client.get_tpm_keys().await?;
+
+                model::DeviceRegistration {
+                    registration_id: Some(registration_id.into()),
+                    tpm: Some(model::TpmAttestation {
+                        endorsement_key: base64::encode(&endorsement_key),
+                        storage_root_key: Some(base64::encode(&storage_root_key)),
```

Comment:
Any reason this is an Option? It is always specified in TpmAttestation

Reply:
Ah, I had copied the definition from the old repo, which probably auto-generated these structures. But yes, we always pass a string here, so I'll go ahead and change the defn